src/Recaptcha.js

```diff
@@ -1,227 +1,260 @@
+          $isClosed.current = false;
         },
         close: handleClose,
-    }), [handleClose]);
+      }),
+      [handleClose]
+    );
 
     const handleNavigationStateChange = useCallback(() => {
-        // prevent navigation on Android
-        if (!loading) {
-            $webView.current.stopLoading();
-        }
+      // prevent navigation on Android
+      if (!loading) {
+        $webView.current.stopLoading();
+      }
     }, [loading]);
 
-    const handleShouldStartLoadWithRequest = useCallback(event => {
+    const handleShouldStartLoadWithRequest = useCallback(
+      (event) => {
         // prevent navigation on iOS
-        return event.navigationType === 'other';
-    }, [loading]);
+        return event.navigationType === "other";
+      },
+      [loading]
+    );
 
-    const webViewStyles = useMemo(() => [
-        styles.webView,
-        style,
-    ], [style]);
+    const webViewStyles = useMemo(() => [styles.webView, style], [style]);
 
     const renderLoading = () => {
-        if (!loading && source) {
-            return null;
-        }
-        return (
-            <View style={styles.loadingContainer}>
-                {loadingComponent || <ActivityIndicator size="large" />}
-            </View>
-        );
+      if (!loading && source) {
+        return null;
+      }
+      return (
+        <View style={styles.loadingContainer}>
+          {loadingComponent || <ActivityIndicator size="large" />}
+        </View>
+      );
     };
 
-    return (
-        <Modal
-            transparent
-            {...modalProps}
-            visible={visible}
-            onRequestClose={handleClose}
-        >
-            {headerComponent}
-            <WebView
-                bounces={false}
-                allowsBackForwardNavigationGestures={false}
-                originWhitelist={originWhitelist}
-                onShouldStartLoadWithRequest={handleShouldStartLoadWithRequest}
-                onNavigationStateChange={handleNavigationStateChange}
-                {...webViewProps}
-                source={source}
-                style={webViewStyles}
-                onMessage={handleMessage}
-                ref={$webView}
-            />
-            {footerComponent}
-            {renderLoading()}
-        </Modal>
+    const renderCaptchaComponent = () => (
+      <View>
+        {headerComponent}
+        <WebView
+          bounces={false}
+          allowsBackForwardNavigationGestures={false}
+          originWhitelist={originWhitelist}
+          onShouldStartLoadWithRequest={handleShouldStartLoadWithRequest}
+          onNavigationStateChange={handleNavigationStateChange}
+          {...webViewProps}
+          source={source}
+          style={webViewStyles}
+          onMessage={handleMessage}
+          ref={$webView}
+        />
+        {footerComponent}
+        {renderLoading()}
+      </View>
     );
-});
+
+    return renderWithoutModal ? (
+      <View>{renderCaptchaComponent()}</View>
```

Comment:
If renderWithoutModal is true, then renders with a View instead of a Modal component.